src/Symfony/Component/Debug/DebugClassLoader.php

```diff
@@ -11,515 +11,13 @@
 
 namespace Symfony\Component\Debug;
 
-use PHPUnit\Framework\MockObject\Matcher\StatelessInvocation;
+use Symfony\Component\ErrorHandler\DebugClassLoader as BaseDebugClassLoader;
+
+@trigger_error(sprintf('The "%s" class is deprecated since Symfony 4.4, use "%s" instead.', DebugClassLoader::class, BaseDebugClassLoader::class), E_USER_DEPRECATED);
 
 /**
- * Autoloader checking if the class is really defined in the file found.
- *
- * The ClassLoader will wrap all registered autoloaders
- * and will throw an exception if a file is found but does
- * not declare the class.
- *
- * @author Fabien Potencier <fabien@symfony.com>
- * @author Christophe Coevoet <stof@notk.org>
- * @author Nicolas Grekas <p@tchwork.com>
- * @author Guilhem Niot <guilhem.niot@gmail.com>
+ * @deprecated since Symfony 4.4, use Symfony\Component\ErrorHandler\DebugClassLoader instead.
  */
-class DebugClassLoader
+class DebugClassLoader extends BaseDebugClassLoader
```

Comment:
I'm not sure if the `DebugClassLoader` really should be moved. To me, the Debug component seems to be a better fit than the new ErrorHandler component.

Reply:
IMHO it fit better in the new component `ErrorHandler`: because it's part of the error mechanism:
> Autoloader checking if the class is really defined in the file found.
> 
> The ClassLoader will wrap all registered autoloaders
> and will throw an exception if a file is found but does
> not declare the class.

Reply:
Done.